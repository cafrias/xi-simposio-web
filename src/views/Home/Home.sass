@import '../../_sass/_constants.sass'

.home-container
  width: 100%
  display: flex
  flex-direction: column
  align-items: center

// HERO LOGO ___________________________________________________________________
.hero-logo-container
  padding: 0 2.618rem
  width: 100%
  margin: 2.618rem 0

.hero-logo
  max-width: 100%

// TIME ________________________________________________________________________
.time-container
  width: 100%
  text-align: center

.time
  position: relative
  top: 8px

.time-container::after
  display: block
  content: ""
  height: 10px
  width: 100%
  background: $dark-gradient
  position: relative
  z-index: 2

// HERO PICS ___________________________________________________________________
.hero-pics
  width: 100%
  height: 6.854rem
  background-image: url('./assets/left.jpg')
  background-position: center
  background-size: cover
  text-align: center

@media (min-resolution: 2dppx)
  .hero-pics
    background-image: url('./assets/left@2x.jpg')

@media (min-resolution: 3dppx)
  .hero-pics
    background-image: url('./assets/left@3x.jpg')

// SOCIAL ______________________________________________________________________
.social
  display: inline-block
  background-color: #fff
  position: relative
  z-index: 3
  transform: scale(1.1, 1.1) skew(15deg)
  height: 100%
  width: 50%
  +shadow-1

.social-container
  transform: scale(0.9, 0.9) skew(-15deg)
  width: 100%
  height: 100%
  display: flex
  justify-content: center

.social__mail
  margin-left: 1rem

// ORGANIZED BY ________________________________________________________________
.organized-by
  position: relative
  z-index: 4
  margin: 0
  width: 100%
  display: flex
  align-items: center
  justify-content: center
  height: 6.854rem
  background-color: green
  font-size: 2.023rem
  color: #fff
  font-weight: 900
  text-transform: uppercase
  background: $light-gradient
  +shadow-0
  +anti-aliased

// ORGANIZED ___________________________________________________________________
.organized-section
  width: 100%

.organized
  height: 6.854rem
  display: flex
  justify-content: space-around
  align-items: center
  width: 100%
  background-image: url('./assets/subtle_dots_@2X.png')
  background-repeat: repeat

// SPONSORS ____________________________________________________________________
.sponsors
  height: 110px
  width: 100%
  background-color: $primary
  display: flex
  flex-direction: column
  align-items: center

.sponsors__header
  display: inline-block
  background-color: #fff
  transform: skew(15deg)
  padding: 1rem 1.618rem
  +shadow-1
  +anti-aliased

.sponsors__title
  margin: 0
  color: $dark-grey
  font-weight: 900
  font-size: 2.023rem
  text-transform: uppercase
  transform: skew(-15deg)
