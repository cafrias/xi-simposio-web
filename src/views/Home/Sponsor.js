import React from 'react'

import './Sponsor.sass'

function Sponsor({logo, alt, small}) {
  return(
    <li className={`sponsors__container ${small ? 'sponsors__container--small' : ''}`}>
      <img className="sponsor__img" src={logo} alt={alt}/>
    </li>
  )
}

export default Sponsor
