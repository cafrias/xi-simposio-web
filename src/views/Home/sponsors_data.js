// IMAGES ______________________________________________________________________
import logo01 from './assets/logo-01.svg'
import logo02 from './assets/logo-02.png'
import logo03 from './assets/logo-03.jpg'

export default [
  {
    id: '01',
    logo: logo01,
    alt: 'Logo Legislatura de la Provincia de Tierra del Fuego'
  },
  {
    id: '02',
    logo: logo02,
    alt: 'Logo Gobierno de la Provincia de Tierra del Fuego'
  },
  {
    id: '03',
    logo: logo03,
    alt: 'Logo Banco de la Provincia de Tierra del Fuego'
  }
]
