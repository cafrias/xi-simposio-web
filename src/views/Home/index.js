// @flow

import React from 'react'

// ASSETS ______________________________________________________________________
import './assets/subtle_dots_@2X.png'
import './Home.sass'

import logo from './assets/logo.svg'
import time from './assets/time.svg'

import idei_x2 from './assets/idei@2x.png'
import untdf_x2 from './assets/untdf@2x.png'

// ICONS _______________________________________________________________________
import FacebookIcon from './FacebookIcon'
import GmailIcon from './GmailIcon'

function Home() {
  return (
    <section className="home-container">
      <div className="hero-logo-container">
        <img className="hero-logo" src={logo} alt="Simposio Ushuaia logo con lema"/>
      </div>
      <div className="time-container">
        <img src={time} alt="17/05 - 19/05" className="time"/>
      </div>
      <div className="hero-pics">
        <div className="social">
          <div className="social-container">
            <a className="social__icon-link" target="_blank" href="https://es-la.facebook.com/people/Simposio-Ushuaia/100008262517328">
              <FacebookIcon />
            </a>
            <a className="social__icon-link social__mail" href="mailto:ponencias.simposio@gmail.com">
              <GmailIcon />
            </a>
          </div>
        </div>
      </div>
      <section className="organized-section">
        <h2 className="organized-by">
          <span className="organized-by__text">Organizado Por</span>
        </h2>
        <div className="organized">
            <img src={idei_x2} alt="Idei Logo" className="organized__logo"/>
            <img src={untdf_x2} alt="UNTDF Logo" className="organized__logo"/>
        </div>
      </section>
      <section className="sponsors">
        <header className="sponsors__header">
          <h2 className="sponsors__title">Auspiciantes</h2>
        </header>
      </section>
    </section>
  )
}

export default Home
