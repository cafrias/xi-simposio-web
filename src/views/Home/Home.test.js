import React from 'react'
import {shallow} from 'enzyme'

import Home from './index'

describe('Home view', () => {
  var homeWrapper

  beforeEach(() => homeWrapper = shallow(<Home />))

  it('should render without errors', () => {
    expect(homeWrapper).toHaveLength(1)
  })

  it('should match snapshot', () => {
    expect(homeWrapper).toMatchSnapshot()
  })
})
