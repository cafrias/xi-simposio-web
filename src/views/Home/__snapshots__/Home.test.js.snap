exports[`Home view should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="home-container">
    <div
        className="hero-logo-container">
        <img
            alt="Simposio Ushuaia logo con lema"
            className="hero-logo"
            src="logo.svg" />
    </div>
    <div
        className="time-container">
        <img
            alt="17/05 - 19/05"
            className="time"
            src="time.svg" />
    </div>
    <div
        className="hero-pics">
        <div
            className="social">
            <SocialLinks />
        </div>
    </div>
    <section
        className="organized-section">
        <h2
            className="organized-by">
            <span
                className="organized-by__text">
                Organizado Por
            </span>
        </h2>
        <div
            className="organized">
            <img
                alt="Idei Logo"
                className="organized__logo"
                src="idei@2x.png" />
            <img
                alt="UNTDF Logo"
                className="organized__logo"
                src="untdf@2x.png" />
        </div>
    </section>
    <section
        className="background_orange sponsors">
        <header
            className="sponsors__header">
            <h2
                className="section__title">
                Auspiciantes
            </h2>
        </header>
        <ul
            className="sponsors__grid">
            <Sponsor
                alt="Logo Legislatura de la Provincia de Tierra del Fuego"
                logo="logo-01.svg" />
            <Sponsor
                alt="Logo Gobierno de la Provincia de Tierra del Fuego"
                logo="logo-02.png" />
            <Sponsor
                alt="Logo Banco de la Provincia de Tierra del Fuego"
                logo="logo-03.jpg" />
            <Sponsor
                alt="Logo Consejo Profesional de Ciencias Económicas de Tierra del Fuego"
                logo="logo-04.gif" />
        </ul>
        <section
            className="collaboration">
            <h3
                className="collaboration__title">
                Se agradece la colaboración de:
            </h3>
            <ul
                className="sponsors__grid collaborators__grid">
                <Sponsor
                    alt="Logo Laguna Negra Chocolatería"
                    logo="logo-05.jpg"
                    small={true} />
                <Sponsor
                    alt="Logo Rumbo Sur SRL"
                    logo="logo-06.png"
                    small={true} />
            </ul>
        </section>
    </section>
</section>,
  "nodes": Array [
    <section
      className="home-container">
      <div
            className="hero-logo-container">
            <img
                  alt="Simposio Ushuaia logo con lema"
                  className="hero-logo"
                  src="logo.svg" />
      </div>
      <div
            className="time-container">
            <img
                  alt="17/05 - 19/05"
                  className="time"
                  src="time.svg" />
      </div>
      <div
            className="hero-pics">
            <div
                  className="social">
                  <SocialLinks />
            </div>
      </div>
      <section
            className="organized-section">
            <h2
                  className="organized-by">
                  <span
                        className="organized-by__text">
                        Organizado Por
                  </span>
            </h2>
            <div
                  className="organized">
                  <img
                        alt="Idei Logo"
                        className="organized__logo"
                        src="idei@2x.png" />
                  <img
                        alt="UNTDF Logo"
                        className="organized__logo"
                        src="untdf@2x.png" />
            </div>
      </section>
      <section
            className="background_orange sponsors">
            <header
                  className="sponsors__header">
                  <h2
                        className="section__title">
                        Auspiciantes
                  </h2>
            </header>
            <ul
                  className="sponsors__grid">
                  <Sponsor
                        alt="Logo Legislatura de la Provincia de Tierra del Fuego"
                        logo="logo-01.svg" />
                  <Sponsor
                        alt="Logo Gobierno de la Provincia de Tierra del Fuego"
                        logo="logo-02.png" />
                  <Sponsor
                        alt="Logo Banco de la Provincia de Tierra del Fuego"
                        logo="logo-03.jpg" />
                  <Sponsor
                        alt="Logo Consejo Profesional de Ciencias Económicas de Tierra del Fuego"
                        logo="logo-04.gif" />
            </ul>
            <section
                  className="collaboration">
                  <h3
                        className="collaboration__title">
                        Se agradece la colaboración de:
                  </h3>
                  <ul
                        className="sponsors__grid collaborators__grid">
                        <Sponsor
                              alt="Logo Laguna Negra Chocolatería"
                              logo="logo-05.jpg"
                              small={true} />
                        <Sponsor
                              alt="Logo Rumbo Sur SRL"
                              logo="logo-06.png"
                              small={true} />
                  </ul>
            </section>
      </section>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Home />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="home-container">
          <div
                    className="hero-logo-container">
                    <img
                              alt="Simposio Ushuaia logo con lema"
                              className="hero-logo"
                              src="logo.svg" />
          </div>
          <div
                    className="time-container">
                    <img
                              alt="17/05 - 19/05"
                              className="time"
                              src="time.svg" />
          </div>
          <div
                    className="hero-pics">
                    <div
                              className="social">
                              <SocialLinks />
                    </div>
          </div>
          <section
                    className="organized-section">
                    <h2
                              className="organized-by">
                              <span
                                        className="organized-by__text">
                                        Organizado Por
                              </span>
                    </h2>
                    <div
                              className="organized">
                              <img
                                        alt="Idei Logo"
                                        className="organized__logo"
                                        src="idei@2x.png" />
                              <img
                                        alt="UNTDF Logo"
                                        className="organized__logo"
                                        src="untdf@2x.png" />
                    </div>
          </section>
          <section
                    className="background_orange sponsors">
                    <header
                              className="sponsors__header">
                              <h2
                                        className="section__title">
                                        Auspiciantes
                              </h2>
                    </header>
                    <ul
                              className="sponsors__grid">
                              <Sponsor
                                        alt="Logo Legislatura de la Provincia de Tierra del Fuego"
                                        logo="logo-01.svg" />
                              <Sponsor
                                        alt="Logo Gobierno de la Provincia de Tierra del Fuego"
                                        logo="logo-02.png" />
                              <Sponsor
                                        alt="Logo Banco de la Provincia de Tierra del Fuego"
                                        logo="logo-03.jpg" />
                              <Sponsor
                                        alt="Logo Consejo Profesional de Ciencias Económicas de Tierra del Fuego"
                                        logo="logo-04.gif" />
                    </ul>
                    <section
                              className="collaboration">
                              <h3
                                        className="collaboration__title">
                                        Se agradece la colaboración de:
                              </h3>
                              <ul
                                        className="sponsors__grid collaborators__grid">
                                        <Sponsor
                                                  alt="Logo Laguna Negra Chocolatería"
                                                  logo="logo-05.jpg"
                                                  small={true} />
                                        <Sponsor
                                                  alt="Logo Rumbo Sur SRL"
                                                  logo="logo-06.png"
                                                  small={true} />
                              </ul>
                    </section>
          </section>
</section>,
        "_debugID": 4,
        "_renderedOutput": <section
          className="home-container">
          <div
                    className="hero-logo-container">
                    <img
                              alt="Simposio Ushuaia logo con lema"
                              className="hero-logo"
                              src="logo.svg" />
          </div>
          <div
                    className="time-container">
                    <img
                              alt="17/05 - 19/05"
                              className="time"
                              src="time.svg" />
          </div>
          <div
                    className="hero-pics">
                    <div
                              className="social">
                              <SocialLinks />
                    </div>
          </div>
          <section
                    className="organized-section">
                    <h2
                              className="organized-by">
                              <span
                                        className="organized-by__text">
                                        Organizado Por
                              </span>
                    </h2>
                    <div
                              className="organized">
                              <img
                                        alt="Idei Logo"
                                        className="organized__logo"
                                        src="idei@2x.png" />
                              <img
                                        alt="UNTDF Logo"
                                        className="organized__logo"
                                        src="untdf@2x.png" />
                    </div>
          </section>
          <section
                    className="background_orange sponsors">
                    <header
                              className="sponsors__header">
                              <h2
                                        className="section__title">
                                        Auspiciantes
                              </h2>
                    </header>
                    <ul
                              className="sponsors__grid">
                              <Sponsor
                                        alt="Logo Legislatura de la Provincia de Tierra del Fuego"
                                        logo="logo-01.svg" />
                              <Sponsor
                                        alt="Logo Gobierno de la Provincia de Tierra del Fuego"
                                        logo="logo-02.png" />
                              <Sponsor
                                        alt="Logo Banco de la Provincia de Tierra del Fuego"
                                        logo="logo-03.jpg" />
                              <Sponsor
                                        alt="Logo Consejo Profesional de Ciencias Económicas de Tierra del Fuego"
                                        logo="logo-04.gif" />
                    </ul>
                    <section
                              className="collaboration">
                              <h3
                                        className="collaboration__title">
                                        Se agradece la colaboración de:
                              </h3>
                              <ul
                                        className="sponsors__grid collaborators__grid">
                                        <Sponsor
                                                  alt="Logo Laguna Negra Chocolatería"
                                                  logo="logo-05.jpg"
                                                  small={true} />
                                        <Sponsor
                                                  alt="Logo Rumbo Sur SRL"
                                                  logo="logo-06.png"
                                                  small={true} />
                              </ul>
                    </section>
          </section>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home />,
}
`;
